### PR TITLE
Issue: https://moderntribe.atlassian.net/browse/TEC-3454

### PR DESCRIPTION
**Pull request by lucatume**
This PR intent is to leverage the improved fully-localized URL
support we've implemented in `release/B20.07` to better support
and interact with WPML.

### CODE REVIEW

File: src/Tribe/I18n.php [hunk 1]
```diff
@@ -173,8 +173,9 @@ public function get_i18n_strings_for_domains( $strings, $languages, $domains = a
 	 * @return mixed The callback return value, if any.
 	 */
 	protected function with_locale( $locale, callable $do, array $args = [] ) {
-		global $wp_filter;
+		global $wp_filter, $l10n;
 		$locale_filters_backup = isset( $wp_filter['locale'] ) ? $wp_filter['locale'] : [];
+		$l10n_backup = $l10n;
```
Comment (tr1b0t) [inline src/Tribe/I18n.php:178]: :warning: Equals sign not aligned with surrounding assignments; expected 11 spaces but found 1 space

<sub>hash: 72bfb376ea34fb9ce0dccda4216969ab</sub>

File: src/Tribe/I18n.php [hunk 2]
```diff
@@ -187,6 +188,8 @@ protected function with_locale( $locale, callable $do, array $args = [] ) {
 
 		// Restore the `locale` filtering functions.
 		$wp_filter['locale'] = $locale_filters_backup;
+		// Restore the previous I10n object.
+		$l10n = $l10n_backup;
```
Comment (tr1b0t) [inline src/Tribe/I18n.php:192]: :no_entry_sign: Overriding WordPress globals is prohibited. Found assignment to $l10n

<sub>hash: ad348e7c9dbf1d3d064f40798bf8d546</sub>

File: src/Tribe/Integrations/WPML/Filters.php [hunk 3]
```diff
@@ -65,8 +66,9 @@ public function filter_tribe_events_rewrite_i18n_slugs_raw( $bases, $method, $do
 		 */
 		$untranslated_bases = array_combine( array_keys( $bases ), array_column( $bases, 0 ) );
 
+		$flags = I18n::COMPILE_STRTOLOWER;
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Filters.php:69]: :warning: Equals sign not aligned with surrounding assignments; expected 12 spaces but found 1 space

<sub>hash: 612f9c819b5e78be711451ddc3fe7f2c</sub>

File: src/Tribe/Integrations/WPML/Utils.php [hunk 4]
```diff
@@ -59,7 +60,9 @@ public static function get_wpml_i18n_strings( array $strings, $locale = null, ar
 		// Get the strings on multiple Domains and Languages
 		// WPML filter is unhooked to avoid the locale being set to the default one
 		remove_filter( 'locale', array( $sitepress, 'locale_filter' ) );
-		$translations = tribe( 'tec.i18n' )->get_i18n_strings_for_domains( $strings, $languages, $domains, $current_locale );
+		$flags = I18n::COMPILE_INPUT;
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Utils.php:63]: :warning: Equals sign not aligned with surrounding assignments; expected 8 spaces but found 1 space

<sub>hash: f59d096c4d18d199caf4bc8f8bb3a27e</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 5]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
 			$category_translation = Tribe__Events__Integrations__WPML__Category_Translation::instance();
 			add_filter( 'tribe_events_category_slug', array( $category_translation, 'filter_tribe_events_category_slug' ), 20, 2 );
 		}
+
+		//// WORK IN PROGRESS!!!
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:88]: :no_entry_sign: No space found before comment text; expected "// // WORK IN PROGRESS!!!" but found "//// WORK IN PROGRESS!!!"

<sub>hash: 9140ae0e08d2a0078d086452f1bc49c0</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 6]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+
+		add_action( "tribe_repository_events_pre_get_posts", function () {
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:90]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: String "tribe_repository_events_pre_get_posts" does not require double quotes; use single quotes instead
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 4c880da5ee160cb0cf4c54216962696a</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 7]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+			$request_uri = isset( $_SERVER['REQUEST_URI'] )
+				? filter_var( $_SERVER['REQUEST_URI'], FILTER_SANITIZE_STRING )
+				: false;
+			if(false === $request_uri){
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:94]: * :no_entry_sign: Space after opening control structure is required
* :no_entry_sign: No space before opening parenthesis is prohibited
* :no_entry_sign: No space after opening parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after IF keyword; 0 found
* :no_entry_sign: There must be a space between if and the first parenthesis '('
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space after opening parenthesis is prohibited
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space before closing parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after closing parenthesis; found 0
* :no_entry_sign: No space before closing parenthesis is prohibited
* :no_entry_sign: You must have a space after a closing parenthesis
* :no_entry_sign: Space between opening control structure and closing parenthesis is required

<sub>hash: 85dc6decd4d29ed43fec735bbcb9b2d1</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 8]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+				return;
+			}
+			$url_language_code = false;
+			global $sitepress;
+			$languages               = array_column( (array) $sitepress->get_active_languages(), 'default_locale' );
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:99]: :warning: Equals sign not aligned with surrounding assignments; expected 12 spaces but found 15 spaces

<sub>hash: fcd35868d22739ffb4671d72f4969894</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 9]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+			$flags = I18n::COMPILE_STRTOLOWER;
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:100]: :warning: Equals sign not aligned with surrounding assignments; expected 16 spaces but found 1 space

<sub>hash: c00eac223ee99f9bb633976d3a40b256</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 10]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+			$translated_bases = tribe( 'tec.i18n' )
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:101]: :warning: Equals sign not aligned with surrounding assignments; expected 5 spaces but found 1 space

<sub>hash: b712288ef80a562ad7c8c43b69d5b456</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 11]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+				->get_i18n_strings( ['archive'=> Tribe__Events__Main::instance()->rewriteSlug], $languages, ['default','the-events-calendar'], 'en_US', $flags );
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:102]: * :no_entry_sign: Missing space after array opener.
* :no_entry_sign: Missing space after array opener.
* :no_entry_sign: Expected 1 space between "'archive'" and double arrow; 0 found
* :no_entry_sign: Expected 1 space before "=>"; 0 found
* :no_entry_sign: Expected 1 space before "=>"; 0 found
* :no_entry_sign: Missing space before array closer.
* :no_entry_sign: Missing space before array closer.
* :no_entry_sign: Missing space after array opener.
* :no_entry_sign: Missing space after array opener.
* :no_entry_sign: Expected 1 space between comma and "'the-events-calendar'"; 0 found
* :no_entry_sign: Expected 1 space between comma and "'the-events-calendar'"; 0 found
* :no_entry_sign: Missing space before array closer.
* :no_entry_sign: Missing space before array closer.

<sub>hash: cc2d4b179996684b20435c56bb2edaac</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 12]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+			$archive_translations = Arr::get($translated_bases,'archive', []);
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:103]: * :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: Expected 1 spaces before closing parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space found after comma in argument list

<sub>hash: 616746bf7c5928720889987bbf71d392</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 13]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+			if(count($archive_translations) !== count($languages)){
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:104]: * :no_entry_sign: Space after opening control structure is required
* :no_entry_sign: No space before opening parenthesis is prohibited
* :no_entry_sign: No space after opening parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after IF keyword; 0 found
* :no_entry_sign: There must be a space between if and the first parenthesis '('
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space after opening parenthesis is prohibited
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: Expected 1 spaces before closing parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: Expected 1 spaces before closing parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space before closing parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after closing parenthesis; found 0
* :no_entry_sign: No space before closing parenthesis is prohibited
* :no_entry_sign: You must have a space after a closing parenthesis
* :no_entry_sign: Space between opening control structure and closing parenthesis is required

<sub>hash: ad1091e713b7b7e85b16cc81ae9a65fd</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 14]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+				return;
+			}
+			$archive_translations = array_combine($languages,$archive_translations);
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:107]: * :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: Expected 1 spaces before closing parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space found after comma in argument list

<sub>hash: dab4f332027dbc3e4f8a4580bb488a77</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 15]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+			foreach ( $archive_translations as $lang => $localized_archive_slug ) {
+				$haystack = urldecode( $request_uri );
+				$needle   = '/' . urldecode( $localized_archive_slug );
+				if ( 0 === strpos( $haystack, $needle ) ) {
+					$url_language_code = $lang;
+					break;
+				}
+			}
+			if( false === $url_language_code){
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:116]: * :no_entry_sign: Space after opening control structure is required
* :no_entry_sign: No space before opening parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after IF keyword; 0 found
* :no_entry_sign: There must be a space between if and the first parenthesis '('
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space before closing parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after closing parenthesis; found 0
* :no_entry_sign: No space before closing parenthesis is prohibited
* :no_entry_sign: You must have a space after a closing parenthesis
* :no_entry_sign: Space between opening control structure and closing parenthesis is required

<sub>hash: e008331fa79a63c0af41dfd2e08a18a2</sub>

File: src/Tribe/Integrations/WPML/WPML.php [hunk 16]
```diff
@@ -80,6 +84,40 @@ protected function hook_filters() {
...
+				return;
+			}
+			do_action( 'wpml_switch_language', $url_language_code );
+		} );
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/WPML.php:120]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 6df965851f8fe6e5ea1480bf84a249eb</sub>

**Comment by lucatume** [comment]
This PR has been deprecated by later work, closing.

**Pull request closed without merging**